code/datums/components/riding.dm

```diff
@@ -0,0 +1,309 @@
+/datum/component/riding
+	var/last_vehicle_move = 0 //used for move delays
+	var/last_move_diagonal = FALSE
+	var/vehicle_move_delay = 2 //tick delay between movements, lower = faster, higher = slower
+	var/keytype
+
+	var/slowed = FALSE
+	var/slowvalue = 1
+
+	var/list/riding_offsets = list()	//position_of_user = list(dir = list(px, py)), or RIDING_OFFSET_ALL for a generic one.
+	var/list/directional_vehicle_layers = list()	//["[DIRECTION]"] = layer. Don't set it for a direction for default, set a direction to null for no change.
+	var/list/directional_vehicle_offsets = list()	//same as above but instead of layer you have a list(px, py)
+	var/list/allowed_turf_typecache
+	var/list/forbid_turf_typecache					//allow typecache for only certain turfs, forbid to allow all but those. allow only certain turfs will take precedence.
+	var/allow_one_away_from_valid_turf = TRUE		//allow moving one tile away from a valid turf but not more.
+	var/override_allow_spacemove = FALSE
+	var/drive_verb = "drive"
+	var/ride_check_rider_incapacitated = FALSE
+	var/ride_check_rider_restrained = FALSE
+	var/ride_check_ridden_incapacitated = FALSE
+
+	var/del_on_unbuckle_all = FALSE
+
+/datum/component/riding/Initialize()
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_MOVABLE_BUCKLE, .proc/vehicle_mob_buckle)
+	RegisterSignal(parent, COMSIG_MOVABLE_UNBUCKLE, .proc/vehicle_mob_unbuckle)
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/vehicle_moved)
+
```

Comment:
wheres the Destroy()

Reply:
There are no variables to clean, nor nothing to be done on Destroy() here.

Reply:
You can probably make it use Registerwithparent and unregisterwithparent such that you can transfer the component when a xeno evolves

Reply:
Done.